src/python/WMCore/WorkQueue/WorkQueue.py

```diff
@@ -816,8 +817,29 @@ def closeWork(self, *workflows):
         self.backend.recordTaskActivity('workclosing', msg)
 
         return workflowsToClose
-
-    def performQueueCleanupActions(self, skipWMBS = False):
+    
+    def deleteCompletedWFElements(self):
+        """
+        deletes Workflow when workflow is in finished status
+        """
+        reqNames = self.backend.getWorkflowNames()
+        reqNamesInbox = self.backend.flowNames(inboxFlag = True)
+        reqNames = list(set(reqNames).union(set(reqNamesInbox)))
+        requestsInfo = self.reqmgrSvc.getRequestByNames(reqNames)
+        deleteRequests = []
+        for key, value in requestsInfo.items():
+            if ((value["RequestStatus"] == None) or \
```

Comment:
But I should add completed, closed-out, failed states as well to clean WQ as soon as possible. I will fix that.